forwardcookie.go

```diff
@@ -55,77 +47,51 @@ func New(ctx context.Context, next http.Handler, config *Config, name string) (h
 }
 
 func (e *ForwardCookie) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
-	isFetchReq := true
-	// if the incoming request contains cookies
-	if len(req.Cookies()) > 0 {
-		// make sure that those cookies won't be added two-fold
-		for _, name := range e.cookies {
-			cookie, err := req.Cookie(name)
-			if err != nil {
-				isFetchReq = false
-			}
-			if cookie != nil {
-				isFetchReq = false
-			}
-		}
+	fetchReq, err := http.NewRequest(http.MethodGet, e.addr, nil)
+	if err != nil {
+		log.Printf("%s", err)
+		return
 	}
-	if isFetchReq {
-		fetchReq, err := http.NewRequest(http.MethodGet, e.addr, nil)
-		if err != nil {
-			log.Fatalf(fmt.Sprintf("%s", err))
-			return
-		}
 
-		AddHeaders(fetchReq, req, e)
-		AddParameters(fetchReq, req, e)
+	addCookiesFromRequest(fetchReq, req, e)
+	addHeaders(fetchReq, req, e)
+	addParameters(fetchReq, req, e)
 
-		client := http.Client{}
-		forwardResponse, err := client.Do(fetchReq)
-		if err != nil {
-			log.Fatalf(fmt.Sprintf("%s", err))
-			return
-		}
+	forwardResponse, err := http.DefaultClient.Do(fetchReq)
+	if err != nil {
+		log.Printf("%s", err)
+		return
+	}
 
-		cookies := Cookies{
-			cookieHeaders: make(map[string]string),
-		}
+	addCookiesFromResponse(rw, forwardResponse, e)
 
-		responseCookies := forwardResponse.Header["Set-Cookie"]
+	e.next.ServeHTTP(rw, req)
+}
 
-		for _, wantedCookie := range e.cookies {
-			for _, header := range responseCookies {
-				cookieName := getCookieName(header)
-				if cookieName == wantedCookie {
-					cookies.cookieHeaders[cookieName] = header
-				}
+// addCookies to rw from resp.
+func addCookiesFromResponse(rw http.ResponseWriter, resp *http.Response, config *ForwardCookie) {
+	for _, wantedCookie := range config.cookies {
+		for _, respCookie := range resp.Cookies() {
+			if respCookie.Name == wantedCookie {
+				rw.Header().Add("Set-Cookie", respCookie.Raw)
 			}
 		}
-
-		for _, cookieHeader := range cookies.cookieHeaders {
-			rw.Header().Add("Set-Cookie", cookieHeader)
-		}
 	}
-
-	e.next.ServeHTTP(rw, req)
 }
 
-// Extract cookie name.
-func getCookieName(setCookieString string) string {
-	parts := strings.Split(textproto.TrimString(setCookieString), ";")
-	if len(parts) == 1 && parts[0] == "" {
-		return ""
-	}
-	cookiePair := textproto.TrimString(parts[0])
-	j := strings.Index(cookiePair, "=")
-	if j < 0 {
-		return ""
+// addCookies to fetchReq from req.
+func addCookiesFromRequest(fetchReq, req *http.Request, config *ForwardCookie) {
```

Comment:
I guess this is nit picking and a bit of personal preference but I do prefer being explicit with types here. What do you think?
```suggestion
func addCookiesFromRequest(fetchReq *http.Request, req *http.Request, config *ForwardCookie) {
```

Reply:
👍